### PR TITLE
Re-add missing CSVRecordReader constructor (with string delim and quote).

### PR DESCRIPTION
**Pull request by turambar**
# What changes were proposed in this pull request?

It's all in the title. @crockpotveggies broke it.

## How was this patch tested?

No tests -- but helps maintain backwards compatibility.

### CODE REVIEW

**Comment by turambar** [comment]
@crockpotveggies

File: datavec-api/src/main/java/org/datavec/api/records/reader/impl/csv/CSVRecordReader.java [hunk 1]
```diff
@@ -98,6 +98,16 @@ public CSVRecordReader(int skipNumLines, char delimiter, char quote) {
         this.csvParser = new SerializableCSVParser(delimiter, quote);
     }
 
+    /**
+     * Skip lines, use delimiter, and strip quotes
+     * @param skipNumLines the number of lines to skip
+     * @param delimiter the delimiter
+     * @param quote the quote to strip
+     */
+    public CSVRecordReader(int skipNumLines, String delimiter, String quote) {
```
Comment (huitseeker) [inline datavec-api/src/main/java/org/datavec/api/records/reader/impl/csv/CSVRecordReader.java]: This ignores `skipNumLines` and probably should not.

Comment (turambar) [inline datavec-api/src/main/java/org/datavec/api/records/reader/impl/csv/CSVRecordReader.java]: Fixed.

**Comment by crockpotveggies** [comment]
This has already been addressed by https://github.com/deeplearning4j/DataVec/pull/360/files I don't understand what's different...?

**Comment by turambar** [comment]
Adding @AlexDBlack so he can sort out the overlap with the other PR.

**Comment by crockpotveggies** [comment]
Alex's PR has already been merged. Are you on an out of date branch? Or are you adding an entirely new constructor?

**Comment by AlexDBlack** [comment]
LGTM, except that I'd probably prefer an ```@deprecated``` on the added constructor... given that only characters are really supported now (not strings), the char delim should be the main constructor IMO. FYI my earlier PR was a different constructor signature (only delim, this one has both delim and quote)

**Pull request merged**